app/dash_app.py

```diff
@@ -1,383 +1,397 @@
-                  'Nazko_3M31': [52.93, -123.73],
-                  'Hoodoo_3M8': [56.77, -131.29],
-                  'Hoodoo_3M11': [56.77, -131.29],
-                  'Hoodoo_3M14': [56.77, -131.29],
-                  'Hoodoo_3M38': [56.77, -131.29],
-                  'Hoodoo_3M41': [56.77, -131.29],
-                  'LavaFork_3M11': [56.42, -130.85],
-                  'LavaFork_3M20': [56.42, -130.85],
-                  'LavaFork_3M21': [56.42, -130.85],
-                  'LavaFork_3M29': [56.42, -130.85],
-                  'LavaFork_3M31': [56.42, -130.85],
-                  'LavaFork_3M41': [56.42, -130.85]}
-    print(centerDict[value])
-    return centerDict[value]
-
-
-# Update backscatter date text from slider
 @app.callback(
-    Output('slider-output-container', 'children'),
-    Input('my-slider', 'value'))
-def update_output(value):
-    selectedDate = dfCohFull.dropna()['Reference Date'].unique()[value]
-    return f'Date: "{selectedDate}"'
+    Output(component_id='coherence-matrix', component_property='figure'),
+    Input(component_id='site-dropdown', component_property='value'))
+def update_coherence(target_id):
+    """Display new coherence matrix."""
+    coherence_csv = _coherence_csv(target_id)
+    print(f'Loading: {coherence_csv}')
+    coherence = _read_coherence(coherence_csv)
+
+    return _plot_coherence(coherence)
 
 
-# Update Backscatter image on leaflet map
 @app.callback(
-    Output('rmlimap', 'layers'),
-    Input('my-slider', 'value'))
-def update_output(value):
-    date = dfCohFull.dropna()['Reference Date'].unique()[value].replace('-',
-                                                                        '')
-    print('cite:{}_HH.rmli.geo.db'.format(date))
-    return 'cite:{}_HH.rmli.geo.db'.format(date)
+    Output(component_id='interferogram-bg', component_property='center'),
+    Input(component_id='site-dropdown', component_property='value'))
+def recenter_map(target_id):
+    """Center map on new site."""
+    coords = TARGET_CENTRES[target_id]
+    print(f'Recentering: {coords}')
+    return coords
+
+
+# @app.callback(
+#     Output('date-display', 'children'),
+#     Input('date-slider', 'value'))
+# def update_output(value):
+#     """Update backscatter date text."""
+#     date = _valid_dates(coherence)[value]
+#     title = f'Date: "{date}"'
+#     print(f'Slider title: {title}')
+#     return title
+
+
+# @app.callback(
+#     Output('intensity-map', 'layers'),
+#     Input('date-slider', 'value'))
+# def update_intensity(value):
+#     """Update backscatter image on map"""
+#     date = _valid_dates(coherence)[value]
+#     date = date.replace('-', '')
+#     layers = f'cite:{date}_HH.rmli.geo.db'
+#     print(f'Updating backscatter: {layers}')
+#     return layers
 
 
 if __name__ == '__main__':
-    app.run_server(host='0.0.0.0', port=8050, debug=False)
+    # TODO login and set up - or at least test - port forwarding
+    # See https://shorturl.at/lnSY1
```

Comment:
Remove ToDo comments and create repo issue

Reply:
Issue exists #12. We can remove the TODO when the issue is cleared.